explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/optim/LimeImpactScoreCalculator.java

```
@@ -0,0 +1,86 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LimeImpactScoreCalculator.class);
+    private static final int TOP_FEATURES = 2;
+
+    @Override
+    public SimpleBigDecimalScore calculateScore(LimeConfigSolution solution) {
+        LimeConfig config = LimeConfigEntityFactory.toLimeConfig(solution);
+        BigDecimal impactScore = BigDecimal.ZERO;
+        List<Prediction> predictions = solution.getPredictions();
+        if (!predictions.isEmpty()) {
+            impactScore = getImpactScore(solution, config, predictions);
+        }
+        return SimpleBigDecimalScore.of(impactScore);
+    }
+
+    private BigDecimal getImpactScore(LimeConfigSolution solution, LimeConfig config, List<Prediction> predictions) {
+        double succeededEvaluations = 0;
+        BigDecimal impactScore = BigDecimal.ZERO;
+        LimeExplainer limeExplainer = new LimeExplainer(config);
+        for (Prediction prediction : predictions) {
+            try {
+                Map<String, Saliency> saliencyMap = limeExplainer.explainAsync(prediction, solution.getModel()).get(
+                        Config.DEFAULT_ASYNC_TIMEOUT, Config.DEFAULT_ASYNC_TIMEUNIT);
+
+                for (Map.Entry<String, Saliency> entry : saliencyMap.entrySet()) {
+                    List<FeatureImportance> topFeatures = entry.getValue().getTopFeatures(TOP_FEATURES);
+                    if (!topFeatures.isEmpty()) {
+                        double v = ExplainabilityMetrics.impactScore(solution.getModel(),
+                                prediction, topFeatures);
+                        impactScore = impactScore.add(BigDecimal.valueOf(v));
+                        succeededEvaluations++;
+                    }
+                }
+            } catch (ExecutionException e) {
+                LOGGER.error("Saliency impact-score calculation returned an error {}", e.getMessage());
```

Is it intentional to only log the error. Shouldn't we rather rethrow the exception?

---

the intention here is not to block the optimization because of one single error only in generating the predictions (transient errors, timeouts in HTTP calls, etc.). 
Of course if all the executions fail then the optimization will not generate a different config.